src/ucp/api/ucp.h

```diff
@@ -721,10 +737,11 @@ ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd);
  *   with @ref UCP_FEATURE_WAKEUP to select proper transport
  *
  * @param [in]  worker    Worker to wait for events on.
+ * @param [in]  adderess  If non-null - will also wait for modifications to the address.
  *
  * @return Error code as defined by @ref ucs_status_t
  */
-ucs_status_t ucp_worker_wait(ucp_worker_h worker);
+ucs_status_t ucp_worker_wait(ucp_worker_h worker, void *address);
```

Comment:
Probably need to support more than one address monitored at the same time... Also need alternative for file-descriptor API to do this. Would you like me to open a separate PR on this so we can discuss it specifically?